packages/datetime/src/dateRangeSelectionStrategy.ts

```diff
@@ -0,0 +1,155 @@
+
+import {
+    areSameDay,
+    DateRange,
+    DateRangeBoundary,
+} from "./common/dateUtils";
+
+export interface IDateRangeSelectionState {
+    /**
+     * The boundary that would be modified by clicking the provided `day`.
+     * May be different from `boundaryToModify` in some special cases
+     * (e.g. when hovering over the other boundary's selected date).
+     */
+    boundary?: DateRangeBoundary;
+
+    /**
+     * The date range that would be selected after clicking the provided `day`.
+     */
+    dateRange: DateRange;
+};
+
+export class DateRangeSelectionStrategy {
+    public static getNextState(currentRange: DateRange,
+                               day: Date,
+                               boundaryToModify?: DateRangeBoundary,
+                               allowSingleDayRange?: boolean) {
```

Comment:
`allowSingleDayRange` doesn't need to be optional here - i guess either swap the arguments or make this an object